src/Entity/Query/Sparql/Query.php

```diff
@@ -299,13 +332,23 @@ protected function result() {
     return $uris;
   }
 
+  /**
+   * {@inheritdoc}
+   */
+  public function sort($field, $direction = 'ASC', $langcode = NULL) {
+    if (!in_array($direction, ['ASC', 'DESC'])) {
```

Comment:
By the way, a minor explanation for assigning the variable here instead of directly passing it as 
`if (!in_array(strtoupper($direction).....`.
Later in this method, the `parent::sort` is being called. This method is again passing the direction in the strtoupper function. 
Now, I don't know how the strtoupper actually compiles so this might be in vain. If it saves a few cycles in calculation, it is a performance increase. Of course even if I am right, it is so insignificant that it doesn't really make any difference, it was just a fancy thing.